### PR TITLE
Refactor docs root input

### PR DESCRIPTION
**Pull request by benjgil**
## Proposed Changes

- change the root docs to accept an object instead of properties array
- add `isApiObject` property to ensure backwards compatibility
- update html env to use the new object param

### CODE REVIEW

File: scopes/html/html/html-docs-app.ts [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+  ...rest
+}: DocsRootProps) {
 
   // should be mapObject<Record<string, any>, Record<string, () => any>>
   // @ts-ignore TODO fix
   const reactCompositions = mapObject(compositions, (key, value) => [key, htmlToReact(value as HTMLElement)]);
 
-  return DocsRoot(Provider, componentId, docs, reactCompositions, ...rest);
+  return DocsRoot({Provider, componentId, docs, compositions: reactCompositions, ...rest});
 }
```
Comment (KutnerUri) [inline scopes/html/html/html-docs-app.ts:19]: you can now forward all of the props (remove Provider, componendId, docs, and pass them through the ...rest)

File: scopes/react/ui/docs-app/examples-overview/examples-overview.tsx [hunk 2]
```diff
@@ -1,11 +1,12 @@
 import classNames from 'classnames';
 import React from 'react';
 import jsxToString from 'jsx-to-string';
+import { Example } from '@teambit/docs';
```
Comment (KutnerUri) [inline scopes/react/ui/docs-app/examples-overview/examples-overview.tsx:4]:
```suggestion
import type { Example } from '@teambit/docs';
```

**Review by KutnerUri — CHANGES_REQUESTED** [review]

File: scopes/html/html/html-docs-app.ts (hunk 1 above)
Comment (benjgil) [reply scopes/html/html/html-docs-app.ts]: I can, but it's less obvious here what props are being passed and I think it's useful for someone seeing this code to know what's being passed on

Comment (benjgil) [reply scopes/html/html/html-docs-app.ts]: (seeing as we cant just pass them all through as the compositions needs to be refactored here before forwarding)

**Review by KutnerUri — APPROVED** [review]

File: scopes/html/html/html-docs-app.ts (hunk 1 above)
Comment (benjgil) [reply scopes/html/html/html-docs-app.ts]: update - implemented the change

**Review by GiladShoham — APPROVED** [review]

**Pull request merged**